### ISSUE TITLE
Security CVEs

### ISSUE THREAD

**Issue by AlbertoEAF** [spotify/futures-extra#54]
Hello :)

Currently there are two security vulnerabilities through the dependencies on **Guava** and **junit** -- https://mvnrepository.com/artifact/com.spotify/futures-extra/4.3.1 .

Do you think you can fix this?

In principle the APIs didn't change much so should be easy to change.

Thanks!

**Comment by AlbertoEAF** [comment]
Closed in #56.

**Comment by caesar-ralf** [comment]
> Fixed in #56 that was merged to master.
> 
> @caesar-ralf , do you know what's missing for a new release with this fix?
> 
> Thank you :)

fixing an issue with the library as the build seems to not be working as expected

**Comment by AlbertoEAF** [comment]
Seems you're hitting the same checkstyle issues I faced locally (even with the master code prior to this MR).

I ended up not touching that for the MR #56 but to solve part of these issues in master I bumped to checkstyle 3.3.0 and then adapted the checkstyle.xml as some features were deprecated (as they didn't working reliably according to the devs), and also needed moving one of the plugins out of the tree rules in checkstyle.xml to its root.
Still, even with that I could never compile under Java 8 nor 17, only Java 11.
